src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -134,6 +136,126 @@ private VariantOverlapAnnotator initializeOverlapAnnotator(final FeatureInput<Va
+        for (final InfoFieldAnnotation annotationType : infoAnnotations) {
+            if (annotationType instanceof ReducibleAnnotation) {
+                ReducibleAnnotation currentASannotation = (ReducibleAnnotation) annotationType;
+                if (annotationMap.containsKey(currentASannotation.getRawKeyName())) {
+                    final List<ReducibleAnnotationData<Object>> annotationValue = annotationMap.get(currentASannotation.getRawKeyName());
+                    final Map<String, Object> annotationsFromCurrentType = currentASannotation.combineRawData(allelesList, annotationValue);
+                    combinedAnnotations.putAll(annotationsFromCurrentType);
+                    //remove the combined annotations so that the next method only processes the non-reducible ones
+                    annotationMap.remove(currentASannotation.getRawKeyName());
+                }
+            }
+        }
+        return combinedAnnotations;
+    }
+
+    /**
+     * Finalize reducible annotations (those that use raw data in gVCFs)
+     * @param vc    the merged VC with the final set of alleles, possibly subset to the number of maxAltAlleles for genotyping
+     * @param originalVC    the merged but non-subset VC that contains the full list of merged alleles
+     * @return  a VariantContext with the final annotation values for reducible annotations
+     */
+    public VariantContext finalizeAnnotations(VariantContext vc, VariantContext originalVC) {
+        final Map<String, Object> VariantAnnotations = new LinkedHashMap<>(vc.getAttributes());
+
+        // go through all the requested info annotationTypes
+        for (final InfoFieldAnnotation annotationType : infoAnnotations) {
+            if (annotationType instanceof ReducibleAnnotation) {
+
+                ReducibleAnnotation currentASannotation = (ReducibleAnnotation) annotationType;
+
+                final Map<String, Object> annotationsFromCurrentType = currentASannotation.finalizeRawData(vc, originalVC);
+                if (annotationsFromCurrentType != null) {
+                    VariantAnnotations.putAll(annotationsFromCurrentType);
+                }
+                //clean up raw annotation data after annotations are finalized
+                VariantAnnotations.remove(currentASannotation.getRawKeyName());
+            }
+        }
+
+        // generate a new annotated VC
+        final VariantContextBuilder builder = new VariantContextBuilder(vc).attributes(VariantAnnotations);
+
+        // annotate genotypes, creating another new VC in the process
+        final VariantContext annotated = builder.make();
+        return annotated;
+    }
+
+    /**
+     * Calculate the final annotation value from the raw data
+     * @param vc -- contains the final set of alleles, possibly subset by GenotypeGVCFs
+     * @return
+     */
+    public VariantContext annotateContextForActiveRegion(final ReferenceContext referenceContext,
```

Comment:
Can you check whether this method is actually needed in GATK4? 

(eg., look at how it's used by `HaplotypeCaller/Mutect2` in GATK 3.x, and see if it's doing anything materially different than the GATK4 versions of those tools currently do. Is the GATK4 `HaplotypeCaller` properly invoking `annotateRawData()`?)